bee-message/src/error.rs

```diff
@@ -0,0 +1,187 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crypto::Error as CryptoError;
+
+use core::fmt;
+
+/// Error occurring when creating/parsing/validating messages.
+#[derive(Debug)]
+#[allow(missing_docs)]
+pub enum Error {
+    Io(std::io::Error),
+    InvalidAmount(u64),
+    InvalidDustAllowanceAmount(u64),
+    InvalidTreasuryAmount(u64),
+    InvalidMigratedFundsEntryAmount(u64),
+    InvalidInputOutputCount(usize),
+    InvalidUnlockBlockCount(usize),
+    InvalidInputOutputIndex(u16),
+    InvalidReferenceIndex(u16),
+    InvalidInputKind(u8),
+    InvalidOutputKind(u8),
+    InvalidEssenceKind(u8),
+    InvalidPayloadKind(u32),
+    InvalidAddressKind(u8),
+    InvalidSignatureKind(u8),
+    InvalidUnlockBlockKind(u8),
+    InvalidAccumulatedOutput(u128),
+    InputUnlockBlockCountMismatch(usize, usize),
+    InvalidParentsCount(usize),
+    DuplicateError,
+    InvalidAddress,
+    MissingField(&'static str),
+    InvalidPayloadLength(usize, usize),
+    MissingPayload,
+    InvalidHexadecimalChar(String),
+    InvalidHexadecimalLength(usize, usize),
+    InvalidIndexationIndexLength(usize),
+    InvalidIndexationDataLength(usize),
+    InvalidMessageLength(usize),
+    InvalidReceiptFundsCount(usize),
+    MilestonePublicKeysNotUniqueSorted,
+    MilestoneInvalidPublicKeyCount(usize),
+    MilestoneInvalidSignatureCount(usize),
+    MilestonePublicKeysSignaturesCountMismatch(usize, usize),
+    InvalidUnlockBlockReference(usize),
+    DuplicateSignature(usize),
+    TransactionInputsNotSorted,
+    TransactionOutputsNotSorted,
+    MigratedFundsNotSorted,
+    RemainingBytesAfterMessage,
+    ParentsNotUniqueSorted,
+    TailTransactionHashNotUnique(usize, usize),
+    SignaturePublicKeyMismatch(String, String),
+    InvalidSignature,
+    InvalidTailTransactionHash,
+    InvalidPowScoreValues(u32, u32),
+    CryptoError(CryptoError),
+}
```

Comment:
It would be nice for this to be ordered.

Reply:
You mean... alphabetically sorted ?

Reply:
Yes, I think it makes it easier to find what you're looking for.

Reply:
Sure, that's going to be a lot of fun :trollface: